spec/controllers/queue_items_controller_spec.rb

```diff
@@ -82,11 +89,71 @@
         expect(user1.queue_items.size).to eq(0)
       end
 
-      it "redirect to queue index" do
+      it "should redirect to queue index" do
         delete :destroy, user_id: user1, id: queue_item2.id
         expect(response).to redirect_to user_queue_items_path(user1)
       end
 
+      it "should reorder position numbers" do
+        user1.queue_items << queue_item1
+        user1.queue_items << queue_item2
+        user1.queue_items << queue_item3
+
+        delete :destroy, user_id: user1, id: queue_item2.id
+        expect(user1.queue_items.reload).to match_array([queue_item1, queue_item3])
+      end
+    end
+
+    describe "POST update" do
+      before do
+        user1.queue_items << queue_item1
+        user1.queue_items << queue_item2
+        user1.queue_items << queue_item3
+      end
+
+      context "valid input" do
+        context "update to position" do
+          it "should rearrange queue_items" do
+            post :update,user_id: user1, queue_items: [{id: 1, position: 3}, {id: 2, position: 1}, {id: 3, position: 2}]
+            expect(user1.queue_items).to match_array([queue_item2, queue_item3, queue_item1])
+          end
+
+          it "should normalize rearrangement where user enters position more than number of queue_items" do
+            post :update,user_id: user1, queue_items: [{id: 1, position: 5}, {id: 2, position: 4}, {id: 3, position: 2}]
+            expect(user1.queue_items.last.position).to eq(3)
+          end
+
+          it "should redirect to queue items index" do
+            post :update,user_id: user1, queue_items: [{id: 1, position: 3}, {id: 2, position: 1}, {id: 3, position: 2}]
+            expect(response).to redirect_to user_queue_items_path(user1)
+          end
+        end
+
+        context "update to rating" do
+          it "should update already existing rating" do
+            post :update,user_id: user1, queue_items: [{id: queue_item1.id, position: 3, rating: 2.0}, {id: 2, position: 1, rating: 5.0}, {id: 3, position: 2, rating: 1.0}]
+            expect(review1.reload.rating).to eq(2.0)
+          end
+        end
+      end
+
+      context "invalid input" do
+        before do
+          post :update,user_id: user1, queue_items: [{id: 1, position: 5}, {id: 2, position: 4.7}, {id: 3, position: 2}]
+        end
+
+        it "should display a flash message" do
+          expect(flash['danger']).to be_present
+        end
+
+        it "should redirect to queue index page" do
```

Comment:
The word `should` should be deleted not added. Its not needed - read more about this http://betterspecs.org/#should